art_tree.go

```diff
@@ -17,40 +20,90 @@ func NewArtTree() *ArtTree {
 	return &ArtTree{root: nil, size: 0}
 }
 
+type Result struct {
+	Key   string
+	Value interface{}
+}
```

Comment:
perhaps, but the key is not exposed.  the only other solution (this) is to make a special result struct so that the receiver can see what the key was.